stan/math/prim/fun/quad_form_sym.hpp

```diff
@@ -32,8 +32,8 @@ inline plain_type_t<EigMat2> quad_form_sym(const EigMat1& A, const EigMat2& B) {
   const auto& A_ref = to_ref(A);
   const auto& B_ref = to_ref(B);
   check_symmetric("quad_form_sym", "A", A_ref);
-  plain_type_t<EigMat2> ret(B_ref.transpose() * A_ref * B_ref);
-  return value_type_t<EigMat2>(0.5) * (ret + ret.transpose());
+  auto* ret = new plain_type_t<EigMat2>(B_ref.transpose() * A_ref * B_ref);
+  return holder(value_type_t<EigMat2>(0.5) * (*ret + ret->transpose()), ret);
```

Comment:
Why the pointer? Will holder know to delete this?

Reply:
That is the main and only reason for using holder at all!

Reply:
Alternatively I could call make_holder, which would do this allocation.

Reply:
`make_holder` needs to move the `ret`, while allocating directly does not. But the move is not that expensive and `make_holder` has a bit nicer syntax. Would you prefer if we used `make_holder` everywhere?

Reply:
Yeah let's `make_holder` to keep the pattern simple if it's not a big overhead or anything.

Reply:
done